app/models/status_message.rb

```diff
@@ -44,6 +44,15 @@ class StatusMessage < Post
     joins(:mentions).where(:mentions => {:person_id => person.id})
   }
 
+  def comments_authors
+    authors = [];
+    query = StatusMessage.comments_authors_for_post(id)
+    query.each do |record|
+      authors.push(record.author_id)
+    end
+    authors.uniq
+  end
```

Comment:
```rb
Person.where(id: comments.select(:author_id).distinct)
```

Return the relation so no query is made until access and the caller can decide on the data to fetch without doing more than one query.